homeassistant/components/flux_led/light.py

```diff
@@ -306,8 +306,10 @@ def _turn_on(self, **kwargs):
         if self._mode == MODE_RGBW:
             if white is None:
                 self._bulb.setRgbw(*color_util.color_hsv_to_RGB(*color))
-            else:
+            elif hs_color is None:
                 self._bulb.setRgbw(w=white)
+            else:
+                self._bulb.setRgbw(*color_util.color_hsv_to_RGB(*color), w=white)
```

Comment:
This is getting hard to read with all the branches. Since all the calls actually end in `setRgbw()` with different amount of `None`, how about this (untested):

```python
        # Preserve current brightness on color/white level change
        if hs_color is not None:
            if brightness is None:
                brightness = self.brightness
            rgb = color_util.color_hsv_to_RGB(
                hs_color[0], hs_color[1], brightness / 255 * 100)
        elif brightness is not None:
            rgb = color_util.color_hsv_to_RGB(
                self._color[0], self._color[1], brightness / 255 * 100)
        else:
            rgb = [None]*3

        self._bulb.setRgbw(*rgb, w=white)
```

CC @autinerd